src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -0,0 +1,423 @@
+      offer.getHostname,
+      portsEnvVars)
+
+    builder.setCommand(commandInfo)
+
+    val containerInfo = computeContainerInfo(podDefinition.podVolumes, container)
+
+    builder.setContainer(containerInfo)
+
+    container.healthCheck.foreach { healthCheck =>
+      builder.setHealthCheck(computeHealthCheck(healthCheck, container.endpoints))
+    }
+
+    builder
+  }
+
+  private[this] def computeExecutorInfo(
+    podDefinition: PodDefinition,
+    portsMatch: PortsMatch,
+    portMappings: Seq[mesos.NetworkInfo.PortMapping],
+    instanceId: Instance.Id): mesos.ExecutorInfo.Builder = {
+    val executorID = mesos.ExecutorID.newBuilder.setValue(f"marathon-${instanceId.idString}")
+
+    val executorInfo = mesos.ExecutorInfo.newBuilder
+      .setType(mesos.ExecutorInfo.Type.DEFAULT)
+      .setExecutorId(executorID)
+
+    executorInfo.addResources(scalarResource("cpus", PodDefinition.DefaultExecutorCpus))
+    executorInfo.addResources(scalarResource("mem", PodDefinition.DefaultExecutorMem))
+    executorInfo.addAllResources(portsMatch.resources.asJava)
+
+    def toMesosLabels(labels: Map[String, String]): mesos.Labels.Builder = {
+      labels
+        .map{
+          case (key, value) =>
+            mesos.Label.newBuilder.setKey(key).setValue(value)
+        }
+        .foldLeft(mesos.Labels.newBuilder) { (builder, label) =>
+          builder.addLabels(label)
+        }
+    }
+
+    if (podDefinition.networks.nonEmpty) {
+      val containerInfo = mesos.ContainerInfo.newBuilder
+        .setType(mesos.ContainerInfo.Type.MESOS)
+
+      // TODO: Does 'DiscoveryInfo' need to be set?
+
+      podDefinition.networks
+        .filter(network => network.name.isDefined && network.mode == raml.NetworkMode.Container)
+        .foreach{ containerNetwork =>
+          val networkInfo = mesos.NetworkInfo.newBuilder
+            .setName(containerNetwork.name.get)
+
+          containerNetwork.labels
+            .map(_.values)
+            .foreach(labels => networkInfo.setLabels(toMesosLabels(labels)))
+
+          networkInfo.addAllPortMappings(portMappings.asJava)
+
+          containerInfo.addNetworkInfos(networkInfo)
+        }
+
+      executorInfo.setContainer(containerInfo)
+    }
+
+    executorInfo.setLabels(toMesosLabels(podDefinition.labels))
+
+    executorInfo
+  }
+
+  private[this] def computeCommandInfo(
+    podDefinition: PodDefinition,
+    instanceId: Instance.Id,
+    container: raml.MesosContainer,
+    host: String,
+    portsEnvVars: Map[String, String]): mesos.CommandInfo.Builder = {
+    val commandInfo = mesos.CommandInfo.newBuilder
+
+    container.exec.foreach{ exec =>
+      exec.command match {
+        case raml.ShellCommand(shell) =>
+          commandInfo.setShell(true)
+          commandInfo.setValue(shell)
+        case raml.ArgvCommand(argv) =>
+          commandInfo.setShell(false)
+          commandInfo.addAllArguments(argv.asJava)
+          argv.headOption.foreach(commandInfo.setValue)
```

Comment:
This seems to be true only for `MesosExec`, not `CommandHealthCheck`. It's been changed there -> `computeCommandInfo`.